iron-fish/ironfish
# Add migration for Wallet 2.0

## Summary

This migration will upgrade the wallet schema from 11 to 13, and prevent needing a database reset.

The migration freezes the stores for version 11, and 13 into the data folder for migration for use by the migration. You need to freeze the stores, and cannot use the stores in the wallet because they could change with a new migration introduced after this.

I've outlined all of the changes occurring in this migration here, https://github.com/iron-fish/ironfish/blob/account-migration/ironfish/src/migrations/data/013-wallet-2/readme.md#summary

## Testing Plan

1. Run migration on a wallet and check the balance afterwards to see if it's the same
2. Check the migration has run on a wallet with multiple accounts, and that it works

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[x] Yes
```
